src/cmd/device-protection.js

```diff
@@ -372,43 +351,46 @@ module.exports = class DeviceProtectionCommands extends CLICommandBase {
-	 * Waits for the device to reboot.
-	 * This method waits for the device to reboot by checking if the device is ready to accept control requests.
-	 * It waits for a maximum of 60 seconds with a 1-second interval.
-	 */
-	async _waitForDeviceToReboot() {
-		const start = Date.now();
-		while (Date.now() - start < REBOOT_TIME_MSEC) {
-			try {
-				await this._delay(REBOOT_INTERVAL_MSEC);
-				this.device = await usbUtils.reopenDevice({ id: this.deviceId });
-				// Waiting for any control request to work to ensure the device is ready
-				await this.device.getProtectionState();
-				break;
-			} catch (error) {
-				// ignore error
-			}
-		}
-	}
-
 	async _delay(ms){
 		return new Promise((resolve) => setTimeout(resolve, ms));
 	}
 
 	/**
-	 * Resets the device and waits for it to restart.
+	 * Attempts to enter Safe Mode to enable operations on Protected Devices in DFU mode.
 	 *
 	 * @async
 	 * @param {Object} device - The device to reset.
 	 * @returns {Promise<void>}
 	 */
 	async _putDeviceInSafeMode() {
-		if (this.device.isInDfuMode) {
+		try {
 			await this.device.enterSafeMode();
-			await this.device.close();
+		} catch (error) {
+			if (error.message === 'Unsupported DfuSe command') {
+				// Device-OS 6.1.1 introduces the DFUSE_COMMAND_ENTER_SAFE_MODE to enter safe mode while in DFU mode
+				// which facilitates sending control request to the device
+				// This error occurs if device-os < 6.1.1. There are a couple of ways to tackle this error:
+				// 1. Update device-os to 6.1.1 and try the same in DFU mode
+				// 2. Use device in normal mode (Recommending this approach to the user)
+				throw new Error(`Unable to run this command in DFU mode on this Device-OS version. Take your device out of DFU mode and try again.${os.EOL}Visit ${chalk.yellow('https://docs.particle.io')} for more information${os.EOL}`);
+			}
+		}
+		// device.enterSafeMode() is ineffective for device-os < 6.1.3 (TBD). However, it does not throw an error.
+		// If device is still in dfu mode, it likely means that this is an older device-os version
+		// and it cannot be put into safe mode. In this case, we can only tell if the device is
+		// Protected or not (we cannot distinguish between Protected and Protected (Service Mode) / Open).
+		// Current Approach:
+		// - Request user to exit DFU mode for accurate status determination
+		// Alternative considerations:
+		// 1. Provide a general response about Protection status (less precise but more permissive)
+		// 2. Implement version-specific handling for a more tailored user experience
+		//		(but firmware version is not available in the device class for dfu devices)
+		this.device = await usbUtils.reopenDevice({ id: this.deviceId });
+		if (this.device.isInDfuMode) {
+			throw new Error('Device Protection commands unavailable in DFU mode for this Device-OS version. Take the device out of DFU mode and try again.');
+		} else {
+			this.device = await usbUtils.reopenInNormalMode( { id: this.deviceId });
```

Comment:
Does reopenInNormalMode work on a protected device? Sergey suggested to reset the device if the safe mode command doesn't work. I said no need to implement this, but I didn't realize that this code already calls a function that resets the device.

Can this entire function be simplified to
1. If device is in DFU, call enter safe mode, ignoring errors.
2. Reopen device in normal mode

Reply:
>Does reopenInNormalMode work on a protected device? 

It will work as long as it does not require a reset. If the device is in DFU mode (protected), then reset is not allowed and hence I cant use `reopenInNormalMode` to put my protected dfu mode device in normal mode.

> Can this entire function be simplified to
> 1. If device is in DFU, call enter safe mode, ignoring errors.
> 2. Reopen device in normal mode

Isn't this already simplified? 
1. If the device is in DFU mode, I am calling `_putDeviceInSafeMode` which calls `enterSafeMode()`. I can ignore the `Unsupported DfuSe command` error. 
2. If device is still in dfu mode, I cannot reopen a protected dfu mode device in normal mode using `reopenInNormalMode`. I must ask the user to take the device out of DFU mode to send any control requests to the device.

Reply:
I didn't think that `particle usb reset` would work on Protected Devices. 
Just tested and,
`particle usb reset` works on Protected Devices in DFU mode.
`particle usb reset` fails on Protected Devices in normal mode.

Reply:
I will make this change